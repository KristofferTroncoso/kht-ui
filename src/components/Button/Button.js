import React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  padding: 10px;
  border-radius: 5px;
  outline: 0;
  background: tomato;
  color: white;
`;

export function Button({children="Default Button", ...rest}) {
  return (
    <StyledButton {...rest}>
      {children}
    </StyledButton>
  )
}