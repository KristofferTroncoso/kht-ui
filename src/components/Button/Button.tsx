import React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  padding: 10px;
  margin: 5px;
  border-radius: 5px;
  border: 0;
  outline: 0;
  box-shadow: 1px 1px 4px #3a3a3a;
  background: linear-gradient(to right, #f2709c, #ff9472);
  color: white;
  cursor: pointer;

  :hover {
    filter: brightness(1.1);
  }
`;

export function Button({children="Default Button", ...rest}) {
  return (
    <StyledButton {...rest}>
      {children}
    </StyledButton>
  )
}