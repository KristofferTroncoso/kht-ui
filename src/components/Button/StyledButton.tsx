import styled, { css } from "styled-components";


export interface Props {
  variety?: "primary" | "danger" | "dashed";
  block?: boolean;
}

const StyledButton = styled.button<Props>`
  outline: 0;
  cursor: pointer;
  text-align: center;
  margin: 5px;
  height: 35px;
  width: 60px;
  border: 0;
  color: slategray;
  border-radius: 8px;
  transition: 0.2s all ease-in-out;
  ::selection {
    background: 0;
  }

  ${props => props.variety === "primary" && primaryStyle};
  ${props => props.variety === "danger" && dangerStyle};
  ${props => props.variety === "dashed" && dashedStyle};
  ${props => props.disabled && disabledStyle};
  ${props => props.block && blockStyle};
`;

export const primaryStyle = css`
  background: dodgerblue;
  color: white;
  border: 0;
  :hover {
    filter: brightness(1.2);
  }
`;

export const dangerStyle = css`
  background: lightgray;
  color: crimson;
  border: 1px solid darkgray;
  :hover {
    background: crimson;
    color: white;
    border: crimson;
  }
`;

export const dashedStyle = css`
  border: 1px dashed darkgray;
  :hover {
    color: dodgerblue;
    border: 1px dashed dodgerblue;
  }
`;

export const disabledStyle = css`
  background: lightgray;
  color: darkgray;
  border-color: darkgray;
  cursor: not-allowed;
  :hover {
    color: darkgray;
    border-color: darkgray;
    background: lightgray;
    filter: none;
  }
`;

export const blockStyle = css`
  width: 100%;
`;

export default StyledButton;
