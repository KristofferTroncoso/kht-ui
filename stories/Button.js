import React from 'react';

export function Button({children}) {
  return (
    <button>
      {children}
    </button>
  )
}

export default Button;