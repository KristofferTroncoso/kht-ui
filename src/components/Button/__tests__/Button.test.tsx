import React from 'react';
import {Button} from '../Button';
import Renderer from 'react-test-renderer';

test('default', () => {
  const wrapper = Renderer.create(<Button />);
  expect(wrapper).toMatchSnapshot()
})