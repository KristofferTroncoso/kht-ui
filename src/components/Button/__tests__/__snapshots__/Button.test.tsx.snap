// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
<button
  className="sc-bdVaJa kvEjbm"
>
  Default Button
</button>
`;
